lib/trento/application/projectors/host_projector.ex

```diff
@@ -64,6 +65,22 @@ defmodule Trento.HostProjector do
     end
   )
 
+  project(
+    %HostRestored{
+      host_id: id
+    },
+    fn multi ->
+      host = Repo.get!(HostReadModel, id)
```

Comment:
Yes, this is a trick, but it's like tricking the framework, the expected way to use the changeset is to pass the actual entity and the list of changes, in this case the deregistered_at.

Using like that, we can trick because the changeset will detected the deregistered at as changed, but I don't feel this is the right thing to do :/ 

Anyway, we can also comment and document this trick, we want to always update the deregistered_at to nil, maybe it's a right compromise idk

Reply:
I don't think the expected way of using the changeset is passing the actual entity (whatever this means).
It simply needs an initial model with the primary keys and the changeset.
We actually do this often without getting the whole dataset.

The unique "trick" here is to set an initial dummy value to the `deregistered_at`, as if you use a `nil` value in the changeset it doesn't apply the change.

Anyway, I don't want to stall the implementation, so feel free to procede as you see best :slightly_smiling_face:

Reply:
A bit late to the party, I relate with @arbulu89's point but I'm afraid that if we apply this to other contexts we risk to also update other fields that accept nil values, like arrays, ending up with an empty array in the end.

I understand the underlying point but at the same time this is really Ecto's hello world, so no issues